solutions/src/main/java/ru/sberbank/school/task08/SerializableManager.java

```diff
@@ -17,17 +19,33 @@ public SerializableManager(@NonNull String filesDirectoryPath) {
 
     @Override
     public void initialize() {
-        throw new UnsupportedOperationException("Implement me!");
+
     }
 
     @Override
-    public void saveGame(String filename, Savable gameState) throws SaveGameException {
-        throw new UnsupportedOperationException("Implement me!");
+    public void saveGame(@NonNull String filename,@NonNull Savable gameState) {
+            try (FileOutputStream fos = new FileOutputStream(filesDirectory + filename);
+             ObjectOutputStream out = new ObjectOutputStream(fos)) {
+            out.writeObject(gameState);
+        } catch (IOException ex) {
+            ex.printStackTrace();
+        }
     }
 
     @Override
-    public Savable loadGame(String filename) throws SaveGameException {
-        throw new UnsupportedOperationException("Implement me!");
+    public Savable loadGame(@NonNull String filename) throws SaveGameException {
+
+        MapState mapState = null;
+
+            try (FileInputStream fis = new FileInputStream(filesDirectory + filename);
+             ObjectInputStream inputStream = new ObjectInputStream(fis)) {
```

Comment:
&#39;try resource&#39; has incorrect indentation level 13, expected level should be 16.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a815968291>*